flink-runtime/src/main/java/org/apache/flink/streaming/api/lineage/DefaultLineageGraph.java

```diff
@@ -33,14 +33,15 @@
 @Internal
 public class DefaultLineageGraph implements LineageGraph {
     @JsonProperty private final List<LineageEdge> lineageEdges;
-    @JsonProperty private final List<SourceLineageVertex> sources;
-    @JsonProperty private final List<LineageVertex> sinks;
+    @JsonProperty private List<SourceLineageVertex> sources;
+    @JsonProperty private List<LineageVertex> sinks;
+    private final Set<SourceLineageVertex> deduplicatedSources;
```

Comment:
Why not use `Set<SourceLineageVertex>deduplicatedSources` instead of having one `source` and one `deduplicatedSources`? I find that the `sources` is built from `deduplicatedSources`

Reply:
Yes, it can be simplified with set only. Thanks for the feedback.